src/store/Store.tsx

```diff
@@ -0,0 +1,116 @@
+import React, {
+  createContext,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { Todo } from '../types/Todo';
+import { FilterType } from '../types/FilterBy';
+import { getTodos } from '../api';
+import { User } from '../types/User';
+
+type TodoContextType = {
+  filteredTodos: Todo[];
+  filter: FilterType;
+  setFilter: (filter: FilterType) => void;
+  todos: Todo[];
+  query: string;
+  setQuery: (text: string) => void;
+  handleChangeQuery: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  isLoading: boolean;
+  selectedTodo: Todo | null;
+  setSelectedTodo: (todo: Todo | null) => void;
+  user: User | null;
+  setUser: (user: User | null) => void;
```

Comment:
Not the best idea to move all states to a context, try to minimize amount of states 

And use only values that will help to avoid prop drilling

https://engineering.deptagency.com/prop-drilling-in-react-solutions-and-trade-offs-2